### ISSUE TITLE
[en][mage][spell/talent] Ignite Bank

### ISSUE THREAD

**Issue by blinkmaster** [Atlantiss/BugTracker#5539]
Ignite bank does not work on our server. All crits ( thus refreshed ignite ) should increase the total bank of ignite if the crit happend before the tick of ignite 
how it should work: TIMESTAMP - SPELL - IGNITE
1 - crit for 10k - 4k ignite
2.99999999sec - crit for 10k - 8k ignite
5.99999999sec - crit for 10k - 12k ignite
thus with possibility to increase the ignite bank indefinetly with the positive winds of crits in between of ticks.

how it works on our server:

1 - crit for 10k - 4k ignite
2.999999sec - crit for 10k - 8k ignite
3sec 2k ignite dmg
5.99999sec - crit for 10k - 8k-2k+4k = 10k ignite
6sec - 4k ignite dmg ( from 1st crit last part and 2k from 4k ignite from second crit)
etc.

i dont have to say it reduces damage of fire mage alot specially due to static ignite munching implemented ( which was not appearing on retail - munching came from server lag so it was not static 0.2 or smth but could vary and even not exist at all in good playing hours etc. )

http://us.battle.net/wow/en/forum/topic/4038617617

http://www.wowhead.com/spell=12654/ignite

**Comment by Jenovas** [comment]
which was not appearing on retail - munching came from server lag so it was not static 0.2 or smth but could vary and even not exist at all in good playing hours etc. - not true, blizzard has a static lag even with 1 player online. They do a "global update" every say 100ms instead of updating every player in real time, that is why even with 50ms it takes a while for your auras to apply to a target, please dont write about stuff you have no idea about.

**Comment by blinkmaster** [comment]
ur right sorry. logged on trial account to check this out and ur right ;) but neverless ppl on blizz forum claimed that ignite munching does not always happen even when 2 crits are taken on the exact same time.
but nevermind, this issue i posted here is about smth else

**Comment by aneq** [comment]
The problem is not related to ignite munching at all. It has absolutely nothing to do with it.
There are two possible sources of the problem. I can't log into my mage atm so I can't rule any of them out. I'll just describe both of them.
Let's see how ignite works : when you crit 40% of the damage is left behind as a damage over time effect. The dot should tick twice over the 4 second period, each tick expending the ignite bank remaining. The first tick should be at 2 seconds and the second tick should be at 0 second, right as the ignite fades from the target.
Let's assume we live in a perfect world, where our fireballs always crit and there's a 1.5ish sec interval between fireballs hitting our target (so simply put, alysrazor encounter). Ignite is refreshed before the 2 second mark, so our ignite should never tick, allowing our ignite bank to grow.
Let's also assume that our fireballs always crit for 25k damage so we always get 10k ignite..

How does that pan out? Let's see

00:000 You crit with fireball for 25k damage, 10k ignite is created
00:015 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 20k ignite bank
00:030 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 30k ignite bank
00:045 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 40k ignite bank
00:060 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 50k ignite bank
00:075 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 60k ignite bank

As you can see, ignite never ticks so the ignite bank keeps growing. It might seem that there is no difference because ignite damage stays the same, just dealt later. The difference is in the combustion. Combustion is based on all your active dot effects on the target. Having 300k ignite banked is having combustion ticking for at least 300k per second for 10 seconds.

Here is what currently happens on Atlantiss™

00:000 You crit with fireball for 25k damage, 10k ignite is created
00:015 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 20k ignite bank
00:020 Ignite ticks for 10k, 10k ignite remaining in the bank
00:030 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 20k ignite bank
00:040 Ignite ticks for 10k, 10k ignite remaining in the bank
00:045 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 20k ignite bank
00:060 Ignite ticks for 5k damage, 5k ignite remaining in the bank
00:060 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 15k ignite bank
00:075 You crit with fireball for 25k damage, 10k is added to the remaining ignite, resulting in 25k ignite bank
00:080 Ignite ticks for 12,5k, 12,5k remaining in the bank

As you can see, because the ignite keeps ticking despite being refreshed constantly, its impossible to build a big ignite bank, resulting in lackluster combustions.

There are two possible reasons for this (I think)

1) Ignite first tick happens too early (first tick happens at ignite application and not at 2 seconds, so its impossible to extend ignite bank before the first tick)
2) When a new ignite is reapplied, the tick timer is not reset, and just keeps going, so it's impossible to build a proper ignite bank.

**Comment by Kejmil** [comment]
if ignite is already present on target (maybe condition is more specific like only more than X s ignite on target will cause this, don't know how it exactly works) crits should make 6s ignite (ticks at 4, 2 and 0 intervals), that is also a reason why 1crit combustion is not so far behind by 2crit combustion:
https://youtu.be/OOo-9p_-AXc?t=20
propably main reason of this bug is that ignite is using dot clipping mechanic so we never have 4s or 6s ignite but always less, resulting in ticks happening before we can possibly crit

**Comment by kadi2** [comment]
If that would be so simple Aneq, you would see combustion ticking for milions and one-shotting the boss in that movie Kejmil posted

**Comment by ponpontos** [comment]
Except that having 100% crit / that amount of haste only happens in that situation, realistically speaking it would be impossible to store up a giant ignite of such proportions outside of Alysrazor. 

Right now, as a 363ilvl fire mage, getting a combustion of 10k average tick is unbelievably hard, you need to have all of your int gain procs perfectly lined up at the same time you get a good pyroblast crit. It's a bit ridiculous and screws up fire mages quite a bit.

**Comment by aneq** [comment]
actually we did some research on this, and neither of us are correct. (outside of the fact that ignite doesn't work like it did on retail). The truth is, Ignite works exactly like it should, but it was bugged on retail.  Theres some dot clipping involved there, I'm not sure exactly how it works, but one of the devs, slamas figured it out in detail. The question is. since similar to ignite munching, blizzard balanced around it instead of fixing it, should this bug be introduced on atlantiss as well?. Either ignite munching needs to be removed, or the ignite bug needs to be implemented.

**Comment by adryjan** [comment]
Just add it like, munching. No half measures.

**Comment by kadi2** [comment]
That ignite munching would need to be reworked to work blizzlike way.

**Comment by Caturix** [comment]
This is really annoying, makes me want to reroll my class because main mechanism for my spec is broken. I've been dpsin the dummies for like 2 hours now, and getting high ignite without int/sp procs is impossible in most cases. On top of that, i'm 375 ilvl mage and i'm on 8k SP self buffed.

**Comment by blinkmaster** [comment]
Yup fire is nerfed very hard by this mechanic

**Comment by Jenovas** [comment]
This issue has nothing to do with Ignite Munching, this is a bug that if a ignite tick happens the same time as a refresh happens the new ignite should not be reduced by the tick that happened.

**Comment by Chorbicus** [comment]
Well guys, I have 2 screenshots that are kinda strange... And made me thinking about all this when I've watched em for few minutes. Somthing is surely wrong here, very wrong. :)
1. 
![wowscrnshot_012716_195844](https://cloud.githubusercontent.com/assets/16925360/12626242/01812dc0-c538-11e5-98c2-39edd6ef3089.jpg)
Really strange things heppening here... But check out the second screenshot I've made.
2.
![wowscrnshot_012716_195845](https://cloud.githubusercontent.com/assets/16925360/12626271/37ea6098-c538-11e5-8fd3-f45d7f7445b8.jpg)
The yellow flowing text says that my Combustion is ticking for 22.9k dmg, but the DoT on the target says totally different amount of dmg/tick.
While on first Screenshot I've added, in open Recount window u can see that Combustion IS ticking for 22.9k dmg every sec. And it counts like that for overal dps. 
And by the way, I'm not sure if im right, but Combustion must Combine 3 periodic fire effects - and if that is right - as u can see on screenshots, we have Living Bomb (6489)+ Ignite (15155) + Pyroblast DoT (4162) = 26.4k Combustion should be applied (and if im right, (+ your mastery rating) mine is 37%).
So its not even close to 22.9k how much I got on this Combustion test here.
To be honest, I'm totally confused.

**Comment by adryjan** [comment]
Ye rly strange, but combustion dont profit from our mastery. The one thing is important, actuall dots (pyro dot, LB) doesn't count. Important are your stats in moment when combustion is casted. It counts LB and Pyro dot in that time, not actuall which is on the target. Here some info from EJ:

> The formula :
> 
> Tick damage = PyroDotPower/3 + LbDotPower/3 + IgniteTickDamage/2
> 
> The Ignite tick damage is self explanatory, it's the value you see happening on your combat log, nothing else. The power of the living bomb DOT and the pyro DOT on the hand are much complicated to calculate. Keep in mind that Combustion only check for the presence of the LB and pyro dot as it will recalculate their power with your current stats at the time of casting Combustion, not the ones you had when casting the DOTs.
> 
> LbDotPower = (234 + (0.258*SpellBonusDamage))*(FirePowerTalentRank)*(1.25)*(CriticalMassTalent)*((0.00015618*MasteryRating) + 1.224)
> 
> PyroDotPower = (164 + (0.180*SpellBonusDamage))*(FirePowerTalentRank)*(1.25)*((0.00015618*MasteryRating) + 1.224)
> 
> Multiply this Tick damage by the number of ticks you can expect depending of your haste, add in a critical chance for each dot, add the front damage from the Combustion spell and you'll have your potential total Combustion damage.

Source: http://forums.elitistjerks.com/forums/topic/128507-433-fire-mage-compendium/

We should do some test without changing stats in combat (no enchants, trinkets etc.).

**Comment by Chorbicus** [comment]
Well, I think its wrong. Since Combustion spell (  http://cata.openwow.com/spell=11129  ), combines DoTs on target, so if your DoTs are 10 sec long, and on 5th second your procs goes off, and u lose Spell power, that only effects the Instant fire damage that Combustion deals in the moment when u cast it.
So actually if your DoTs are like mine up there on the photo, my Combustion should be over 26k dps, not 22k like it is. 
Read every spell how it works, since I've been playing on Retail on these Patches, I can say its totally broken. 
Once I figure out how to upload a Video, where you can see that my Combustion (used with all procs, max stats), Is lower then ACTUAL Ignite on the Target at the same moment.

Best Regards!

**Comment by Jenovas** [comment]
Chorbicus no. Combustion doesnt combine the DMG of all DoTs it combines the DPS of all DoTs.
So say Ignite ticks every 2 seconds for 10k, Pyro ticks every 3 secs for 10k, LB ticks every 3 secs for 10k. You dont add 10k+10k+10k = 30k Combustion. 

You do it like this:
Ignite does 10k every 2 sec = 5k DPS
Pyro does 10k every 3 sec = 3333 DPS
LB does 10k every 3 sec = 3333 DPS
Together we get 11-12k DPS. This means Combustion will do 11-12k DPS. Since Combustion ticks every 1 that means every tick should deal 11-12k DMG.

**Comment by Chorbicus** [comment]
Yeah, thats a logic I was thinking of also... But still I cant get over one simple thing, since I was using Combustion Helper (everyone did when Cataclysm update was released), without that addon, Fire Mage is unplayable.
On offical Blizzard game, while I was playing these patches, all the numbers from Combustion Helper displayed, made the result of used Combustion on target, so the Math and calculation scripts were right there. And, that was the reason why people made Combustion Helper, u can actually calculate your Combustion (if you're fast enough and have no latency). 
I'm totally confused.

**Comment by Jenovas** [comment]
https://www.youtube.com/watch?v=s_QLH57bnQc
Here you see a video of Combustion Helper. The numbers are 6k LB, 10k ignite, 3k Pyroblast but the Combustion he uses does 7500 DMG per Tick. So no Combustion does not add up all DoTs and no you cannot quickly add up numbers from Combustion Helper.

**Comment by Chorbicus** [comment]
Oooukay! :) But let me see, what do you have to say about this link, which was already posted in one of a comments above? :)
https://www.youtube.com/watch?v=OOo-9p_-AXc&feature=youtu.be&t=20

Watch it in max quality and tell me your opinion. And by the way, its from Retail. (not some private random script)

**Comment by Jenovas** [comment]
My video obviously shows that Combustion uses DPS and not DMG. Also I think ALL guides out there state that it also takes DPS.

About why we see in this video huge Combustions:
"A cool tip about Combustion; it effectively double-dips its damage on targets that take increased damage, such as Halfus, Maloriak"
This would explain a bit, I would need to do more research and I believe that Combustion on our server does not double dip from DMG Increase effects.

**Comment by Chorbicus** [comment]
This Formula is actually totally RIGHT! @Jenovas 
There are no such a things like "DPS" ,ethier in Combustion spell ingame description. So it must be the Ticks that it multiply.
If thats the DPS that have to be multiplied with poping Combustion, then its smarter to let Ignite burn - it can make more damage. O_o
The formula :

Tick damage = PyroDotPower/3 + LbDotPower/3 + IgniteTickDamage/2

The Ignite tick damage is self explanatory, it's the value you see happening on your combat log, nothing else. The power of the living bomb DOT and the pyro DOT on the hand are much complicated to calculate. Keep in mind that Combustion only check for the presence of the LB and pyro dot as it will recalculate their power with your current stats at the time of casting Combustion, not the ones you had when casting the DOTs.

LbDotPower = (234 + (0.258SpellBonusDamage))(FirePowerTalentRank)(1.25)(CriticalMassTalent)((0.00015618MasteryRating) + 1.224)

PyroDotPower = (164 + (0.180SpellBonusDamage))(FirePowerTalentRank)(1.25)((0.00015618*MasteryRating) + 1.224)

Multiply this Tick damage by the number of ticks you can expect depending of your haste, add in a critical chance for each dot, add the front damage from the Combustion spell and you'll have your potential total Combustion damage.

Source: http://forums.elitistjerks.com/forums/topic/128507-433-fire-mage-compendium/

This part is the most iteresting...
*Multiply this Tick damage by the number of ticks you can expect depending of your haste, add in a critical chance for each dot, add the front damage from the Combustion spell and you'll have your potential total Combustion damage.*

**Comment by Chorbicus** [comment]
And one more thing please... I'd like You @Jenovas , to check out my video made yesterday, watch its every second, and analyze every second of the content u see... I hope it can help u in those experiments. 
https://www.youtube.com/watch?v=vP_4xXzYunc

I'm not trying to be offensive, just trying to help somehow.
Hope It'll help! 
Peace!

**Comment by Jenovas** [comment]
Your video showed us that Combustion works the way it should. It's DMG is exactly what it should be.

The only bugs that we may have is:
1. Ignite not "carrying" the tick that happened exactly when Ignite gets refreshed
2. Combustion not double diping from DMG Increase aura's on a boss

**Comment by Chorbicus** [comment]
If You say so... Then it must be right. :) But numbers are numbers. 
Farewell.

**Comment by ponpontos** [comment]
You're absolutely wrong Chorbicus, the way the combustion tick is calculated right now is absolutely correct. 

The problem is related to lack of ignite "banking" (or whatever you want to call it), because refreshed ignite ticks are simply resulting in an extended ignite duration or the last ignite tick being basically eaten by the new one. During heroism, we should be getting way bigger combustions than we currently are. The Alysrazor video you linked shows the situation perfectly.

**Comment by Chorbicus** [comment]
Well may be. I am wrong. Then make that Ignite work as it should, so we can test it again. Maybe the Ignite is the root of the whole Combustion problem.
The only part I cant get through is that my Ignite is Higher then the Combustion DoT - that actually sums up all the dots on target.

**Comment by ponpontos** [comment]
Mate, of course your ignite is going to be bigger than Combustion dot. You're supposed to divide your Ignite tick by 2, and if your ignite is not doing more than double the tick of tb+pyro you shouldn't be casting combustion. 

I believe you can see the issue in this screenshot, as you can see there's a 6sec ignite on the dummy. This shouldn't be happening. Also, on a different note to Jenovas, I'd like to bring into attention #6076. I apologize if you're already aware of the issue.

![wowscrnshot_012816_184140](https://cloud.githubusercontent.com/assets/16211262/12655088/261b5748-c5ef-11e5-8bc4-3b156166ce5e.jpg)

**Comment by kadi2** [comment]
As you can see on your screen and my picture above, your fireball cast time is 2.1 seconds. 

On retail you had 2 seconds, which is time between ignite ticks + Time between hitting spell and applying ignite debuff.

That also helped alot in cummulating your ignite bank. You are just not even able to do it with fireball cast time higher than 2 seconds. Adding diff/net latency and the fact there is no ability queue system just makes it impossible to do.

**Comment by Chorbicus** [comment]
Yes. I see. Ignite is broken for sure, and well yeah, if Combustion works like its suposed to, maybe it can miss calculate becouse of Ignite Bank. I really dont know, and I'm totally confused.
I'm not a Developer so I cant see anything thats heppening, as we say 'behind the scenes'... :(

**Comment by ponpontos** [comment]
And huh, messing around a bit more I just saw something really weird happening 
![wowscrnshot_012816_185552](https://cloud.githubusercontent.com/assets/16211262/12655497/218e7398-c5f1-11e5-882e-08e2a43c222f.jpg)
45k ignite
![wowscrnshot_012816_185553](https://cloud.githubusercontent.com/assets/16211262/12655502/26173c74-c5f1-11e5-9900-c9e8a8cb2120.jpg)
11k combustion. 

I stopped casting immediately after getting the ignite. What the heck?

Also, @Kadi, I got that overlapping with fireball crit immediately followed by Pyroblast!, not 2 Fireball crits in a row.

**Comment by Chorbicus** [comment]
Mess around more, and You will find even more wierd things heppening :D As I said already ,I'm totally confused. :(

**Comment by Chorbicus** [comment]
Things that heppens also, I get 22k ignite (ofc RNJesus), whn my Combustion is on CD. And I have Impact on, so lets share that Ignite of 22k with other mobs, let them feel it why not, BOOM, Ignite droped on 6k and shared 6k ticks on the rest of mobs... :O

**Comment by Caturix** [comment]
Shouldn't this have some kind of priority, it's like main mechanism for pve mages.

**Comment by Kejmil** [comment]
like snapshoted, properly working ghoul in frost ( ͡° ͜ʖ ͡°)

**Comment by Jenovas** [comment]
Shouldn't this have some kind of priority, it's like main mechanism for pve mages. - mages are doing fine without this. If mages did say 10k DPS this would be high priority, but if with this bug, mages can still hit the top of the charts by just adjusting their gameplay a little bit, that means this bug is med/low priority.

**Comment by anxanas** [comment]
@Kejmil is right obviously snapshot of ghoul in frost/blood should have higer prioirty =))) Stupid mages just go arcane =)

**Comment by adryjan** [comment]
@anxanas lol, die

**Comment by blinkmaster** [comment]
http://i.imgur.com/838LR.gif

here you go, thats how it should work

**Comment by Zhalox123** [comment]
I dont care w/e the problem is combustion currently in pvp is unbalanced and does too low dmg compared to retail I dont know if its becouse of resilliance scale or becouse ignite or any other dot is influencing not properly this matter should be checked and fixed thx!

**Comment by blinkmaster** [comment]
Bump. According to source i provided we can see that making consequtive crits in beetween ignite ticks interval not only should stack out but also work as 4 sec ignite thus increasing formula for combust further ( for example 25k/2 thus 12.5 dps and not 25k/3 thus 8.334 dps )

**Comment by aneq** [comment]
this issue seems to be fixed

**Comment by blinkmaster** [comment]
Making consq crits in between 2 sec tick time increase ignite without making it 6 sec?

**Comment by aneq** [comment]
i dont know how it works exactly, all i know that if your cast time is below 2 seconds and you have 100% crit chance (and you don't move at all so the time between your crits is exactly the same) ignite will just keep growing

**Comment by blinkmaster** [comment]
Cool :3 i will test this when get back sunday and close if it works

**Comment by adryjan** [comment]
Ye pretty damn good, 200k on alys without special build looks nice.
